frappe/desk/doctype/bulk_update/bulk_update.js

```diff
@@ -3,6 +3,16 @@
 
 frappe.ui.form.on('Bulk Update', {
 	refresh: function(frm) {
+		frm.set_query("document_type", function() {
+			return {
+				filters: [
+					['DocType', 'issingle', '=', 0],
+					['DocType', 'name', 'not in', frappe.model.core_doctypes_list],
+					['DocType', 'restrict_to_domain', 'in', frappe.boot.active_domains]
```

Comment:
Most of the doctypes do not have `restrict_to_domain` set. This means those doctypes are not restricted to any domain or allowed under every domain.

This filter will remove those doctypes.